python-sdk/src/astro/databases/mysql.py

```diff
@@ -0,0 +1,228 @@
+from __future__ import annotations
+
+import pandas as pd
+import sqlalchemy
+from airflow.exceptions import AirflowException
+from airflow.providers.mysql.hooks.mysql import MySqlHook
+from MySQLdb import OperationalError
+
+from astro.constants import DEFAULT_CHUNK_SIZE, LoadExistStrategy, MergeConflictStrategy
+from astro.databases.base import BaseDatabase
+from astro.options import LoadOptions
+from astro.settings import MYSQL_SCHEMA
+from astro.table import BaseTable, Metadata
+from astro.utils.compat.functools import cached_property
+
+DEFAULT_CONN_ID = MySqlHook.default_conn_name
+
+
+class MysqlDatabase(BaseDatabase):
+    DEFAULT_SCHEMA = MYSQL_SCHEMA
+
+    _create_schema_statement: str = (
+        "CREATE SCHEMA IF NOT EXISTS {} "
+        "DEFAULT CHARACTER SET = 'utf8' DEFAULT COLLATE = "
+        "'utf8_unicode_ci'"
+    )
+
+    def __init__(
+        self,
+        conn_id: str = DEFAULT_CONN_ID,
+        table: BaseTable | None = None,
+        load_options: LoadOptions | None = None,
+    ):
+        super().__init__(conn_id)
+        self.table = table
+        self.load_options = load_options
+
+    @cached_property
+    def hook(self) -> MySqlHook:
+        """Retrieve Airflow hook to interface with the mysql database."""
+        conn = MySqlHook(mysql_conn_id=self.conn_id).get_connection(self.conn_id)
+        kwargs = {}
+        if conn.schema is None:
+            if (
+                self.table
+                and self.table.metadata
+                and self.table.metadata.database
+                and self.table.metadata.schema
+            ):
+                raise AirflowException(
+                    "You have provided both database and schema."
+                    "Enter only schema while connecting to MySQL!"
```

Comment:
@phanikumv Why is this a problem?

Reply:
In MySQL, schema is synonymous to database, unlike MSSQL where both are different. So we dont want user to specify same thing multiple times, the airflow hook also supports entering only schema- hence want to make it aligned.